datamade/just-spaces
# Define interface for selecting Census regions

## Overview

This PR sketches out the UI for selecting Census regions when creating new CensusAreas. Also, fix a bug where `View.get_queryset_for_agency()` improperly caches the queryset.

Connects #221, closes #223.

## Notes

- I wanted to get this PR in before actually adding new CensusRegions so that we could figure out what the interface should look like. Because of this, the new view won't really show any new regions -- it should simply illustrate what the user flow will look like, with Philadelphia as the only currently-supported region.

- #221 suggests that we might try to create a "shortcut" workflow that will skip the region selection step for power users who only create CensusAreas in one region. I didn't add that as part of this PR yet but I'd be curious to hear if you have thoughts about how we might achieve that -- there doesn't seem to me to be anywhere obvious where it would make sense.

## Testing Instructions

* Run migrations and start dev server
* Log in and navigate to http://localhost:8000/census-areas/
* Hit the `Create new census area` button
* On the new form view, add a name and agency for your area, and choose `Philadelphia` from the Region selector
* Confirm you get redirected to an updated form including the map with census tracts in Philadelphia
* Go back to http://localhost:8000/census-areas/ and edit an existing CensusArea
* Confirm that the Region input is disabled
